### PR TITLE
Record an event on health check failures.

### PR DESCRIPTION
**Pull request by vmarmol**
This aids in debugging and understanding whether/how the health check
failed.

/cc @yujuhong @dchen1107

### CODE REVIEW

**Comment by dchen1107** [comment]
LGTM

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -1401,6 +1401,12 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, runningPod kub
 			containersToKeep[containerID] = index
 			continue
 		}
+
+		// Record an event for the healthcheck failure.
+		ref, err := kubecontainer.GenerateContainerRef(pod, &container)
+		if err == nil {
+			dm.recorder.Eventf(ref, "failed", "Healthcheck failed with probe result: %v", result)
+		}
```
Comment (yujuhong) [inline pkg/kubelet/dockertools/manager.go]: We also record events sometimes in `prober.Probe`. We should do it in one place.

It'd also be good to distinguish between liveness check and readiness check for debugging (as mentioned in #8238). Perhaps we should consolidate the two PR? What do you think?

Comment (vmarmol) [inline pkg/kubelet/dockertools/manager.go]: Ah good call, I forgot about @thockin's PR. We should definitely merge both of these (more likely use his which I think subsumes this one)

**Comment by vmarmol** [comment]
Closing in favor of #8238

**Pull request closed without merging**